psfr/psfr.py

```diff
@@ -403,6 +413,48 @@ def centroid_fit(data, model, mask=None, variance=None, oversampling=1):
         required shift in units of pixels in the data such that the model matches best the data
     """
 
+    def getProbability(pars, variance, mask, oversampling, model, data):
+
+        model_shifted = shift_psf(psf_center=model, oversampling=oversampling, shift=pars, degrade=True,
+                                  n_pix_star=len(data))
+
+        amp = _linear_amplitude(data, model_shifted, variance=variance, mask=mask)
+
+        chi2 = -1 * np.sum((data - model_shifted * amp) ** 2 / variance * mask)
+        return chi2
+
+
+    ###this function is better if the initial guess (0,0) is far from the center of the pixel.
+    # ##The starting step valueis optimized to my particular pixel size/PSF width ratio,
+    # ##so we probably want someone to be ablte to pass a variable here for more general use.
+
+    def _minimize_emcee(x, data, model, variance, mask, oversampling, nwalkers=25, niter=100, startStep=0.1,
+                        nrestart=2):
+        import emcee
```

Comment:
emcee does not appear in the requirements of the package

Reply:
I would actually rather use the PSO. You can use the one implemented in lenstronomy (then no other dependancies are needed). You find it e.g. here https://github.com/lenstronomy/lenstronomy/blob/11e2733cbabd5b8565adf31a92a135b94a19cee6/lenstronomy/Sampling/sampler.py#L103

There is also no need to re-start the sampler. I think it's going to be much more efficient